src/Common/src/System/Text/ValueStringBuilder.cs

```diff
@@ -21,7 +21,18 @@ public ValueStringBuilder(Span<char> initialBuffer)
             _pos = 0;
         }
 
-        public int Length => _pos;
+        public int Length
+        {
+            get => _pos;
+            set
+            {
+                if (value > _chars.Length)
+                {
+                    Grow(value - _chars.Length);
+                }
+                _pos = value;
```

Comment:
This looks potentially buggy.  Growing the builder might end up getting an array from the pool, which might contain garbage data, but we're updating the position to indicate that it's valid.

Reply:
StringBuilder appends nulls, I'll do that.